### PR TITLE
Add a setting to Enable/Disable Utop usage

### PR DESCRIPTION
**Pull request by OscarLahaie**
I have modified package settings and repl script to let the user decides if he wants to use utop or not.

### CODE REVIEW

**Comment by OscarLahaie** [comment]
This should now work without the errors

**Comment by mnxn** [comment]
We already have a few settings that configure the REPL: `ocaml.repl.path` and `ocaml.repl.args`.
Since this option is only for the REPL, it should have the same prefix. It should be `ocaml.repl.useUtop`.

You should also mention that this is for the REPL in the `description` field.

**Comment by OscarLahaie** [comment]
It should be good now 😉

File: src/repl.ml [hunk 1]
```diff
@@ -20,7 +20,7 @@ let ocaml_utop_setting =
   Settings.create_setting ~scope:Global ~key:"ocaml.useUtop"
     ~of_json:Jsonoo.Decode.bool ~to_json:Jsonoo.Encode.bool
 
-let use_utop () = Settings.get ocaml_utop_setting
+let use_utop = Settings.get ocaml_utop_setting
```
Comment (mnxn) [inline src/repl.ml:23]: You should keep the `()` argument to make it a function. Without the argument, the setting is only read once and the user would have to restart the extension after changing the utop setting.

Comment (OscarLahaie) [reply src/repl.ml]: Done

**Comment by OscarLahaie** [comment]
Why are some tests failing ?

**Comment by mnxn** [comment]
There was a trailing comma in `package.json` that caused an error. I fixed it.
If the new checks don't fail, I will merge the PR.

**Comment by OscarLahaie** [comment]
It seems that i still have a type error

**Comment by mnxn** [comment]
> It seems that i still have a type error

Indeed. `Settings.get` returns a `bool option` in this case. You should handle the `None` possibility in the body of `use_utop` so it returns just a `bool`.

**Comment by OscarLahaie** [comment]
> > It seems that i still have a type error
> 
> Indeed. `Settings.get` returns a `bool option` in this case. You should handle the `None` possibility in the body of `use_utop` so it returns just a `bool`.

Should i return a `false` if i get a `None` ?

**Comment by mnxn** [comment]
I think it should be `true` since that's the default in `package.json`.

**Comment by OscarLahaie** [comment]
> I think it should be `true` since that's the default in `package.json`.

Ok i modify this

**Comment by OscarLahaie** [comment]
Now it is not a problem from my additions normally

**Comment by mnxn** [comment]
Thanks!

**Pull request merged**